src/plugins/iframeShare/src/ShareTab.vue

```diff
@@ -0,0 +1,204 @@
+            <label style="pointer-events: none">
+              {{ $t("IframeSharePlugin.expiration_date") }}
+            </label>
+          </template>
+        </Datepicker>
+
+        <BIMDataCheckbox
+          :disabled="false"
+          :text="$t('IframeSharePlugin.keep_camera_setup')"
+          v-model="keepCameraSetup"
+        />
+      </div>
+
+      <BIMDataButton color="primary" fill radius @click="share">
+        <BIMDataIcon name="chevron" size="xxxs" margin="0 6px 0 0" />
+        <span>{{ $t("IframeSharePlugin.link_creation") }}</span>
+      </BIMDataButton>
+    </template>
+
+    <template v-else>
+      <div class="share-tab__body">
+        <h2>
+          <BIMDataIcon name="share" size="s" margin="0 6px 0 0" />
+          <span>{{ $t("IframeSharePlugin.share") }}</span>
+        </h2>
+        <div class="share-tab__link">
+          <span class="share-tab__link__url">
+            {{ currentShare }}
+          </span>
+          <BIMDataButton
+            class="share-tab__link__btn"
+            height="10px"
+            color="primary"
+            fill
+            square
+            @click="copyUrl"
+          >
+            <span>{{ $t("IframeSharePlugin.copy") }}</span>
+          </BIMDataButton>
+        </div>
+
+        <h2>&lt;&sol;&gt; iframe</h2>
+        <div class="share-tab__iframe">
+          <pre>{{ iframeString }}</pre>
+        </div>
+      </div>
+
+      <BIMDataButton color="primary" fill radius @click="copyIframe">
+        {{ $t("IframeSharePlugin.copy") }}
+      </BIMDataButton>
+    </template>
+  </div>
+</template>
+
+<script>
+import Datepicker from "vuejs-datepicker";
+import fr from "vuejs-datepicker/dist/locale/translations/fr";
+import en from "vuejs-datepicker/dist/locale/translations/en";
+import {
+  BIMDataButton,
+  BIMDataCheckbox,
+  BIMDataIcon,
+} from "@bimdata/design-system/components.js";
+import shareIcon from "../assets/share.svg";
+
+export default {
+  components: {
+    BIMDataButton,
+    BIMDataCheckbox,
+    BIMDataIcon,
+    Datepicker,
+  },
+  props: {
+    shareBackendUrl: {
+      type: String,
+    },
+  },
+  data() {
+    return {
+      fr,
+      en,
+      shareIcon,
+      keepCameraSetup: false,
+      expirationDate: null,
+      currentShare: null,
+    };
+  },
+  computed: {
+    iframeString() {
+      return (
+        `<iframe\n` +
+        `  width="560" height="315"\n` +
+        `  src="${this.currentShare}"\n` +
+        `  frameborder="0"\n` +
+        `  allowfullscreen>\n` +
+        `</iframe>`
+      );
+    },
+  },
+  methods: {
+    async share() {
+      const body = {
+        ifc_ids: this.$viewer.state.ifcs.map(ifc => ifc.id),
+        name: this.shareName,
+        locale: this.$i18n.locale,
+        expires_at: "2021-04-22T06:00:00Z",
+      };
+      if (this.expirationDate) {
+        body.expires_at = this.expirationDate.toISOString();
+      } else {
+        let date = new Date();
+        date.setFullYear(2099);
```

Comment:
this should be set in backend imho

Reply:
OK, were can I findd the backend source code to have a look ?

Reply:
https://github.com/bimdata/share-back